README.md

```diff
@@ -0,0 +1,843 @@
+---
+title: Testing workflow for web-components
+published: false
+description: Testing and debugging your web component is an important skill to give you the confidence the code you share is production ready.
+tags: webcomponents, javascript, testing, karma
+---
+
+Whenever you ship something to be used by others you take on a responsibility.
+Unfortunately not always is this responsibility taken care of.
+One thing to make sure you take it on is to write tests.
+
+No matter how small - no matter how simple there should be tests.
+*Yes I know reality hit's hard and there will be many cases where that doesn't happen - but strife for it*
+
+### Disclaimer
```

Comment:
What if you create an input wrapper component that takes a user provided input element as a slot? That aligns to best practices and lets you demonstrate light dom.

Additionally maybe it is more powerful to start with an elemeny with only shadow dom, and show a different element with light dom later. That reduces the barrier of entry which is very important. If we confuse people in the start they wont continue

Reply:
this post is definitely more for the advanced users => I added a warning saying explaining what to expect and what you should know before...